frr-metrics/collector/bfd_test.go

```diff
@@ -0,0 +1,157 @@
+	# TYPE metallb_bfd_echo_packet_output counter
+	metallb_bfd_echo_packet_output{peer="{{ .Peer }}"} {{ .EchoPacketOutput }}
+	# HELP metallb_bfd_session_down_events Number of BFD session down events
+	# TYPE metallb_bfd_session_down_events counter
+	metallb_bfd_session_down_events{peer="{{ .Peer }}"} {{ .SessionDownEvents }}
+	# HELP metallb_bfd_session_up_events Number of BFD session up events
+	# TYPE metallb_bfd_session_up_events counter
+	metallb_bfd_session_up_events{peer="{{ .Peer }}"} {{ .SessionUpEvents }}
+	# HELP metallb_bfd_zebra_notifications Number of BFD zebra notifications
+	# TYPE metallb_bfd_zebra_notifications counter
+	metallb_bfd_zebra_notifications{peer="{{ .Peer }}"} {{ .ZebraNotifications }}
+	`
+
+	bfdTests = []struct {
+		desc                string
+		vtyshOutput         string
+		peer                string
+		sessionUp           int
+		controlPacketInput  int
+		controlPacketOutput int
+		echoPacketInput     int
+		echoPacketOutput    int
+		sessionUpEvents     int
+		sessionDownEvents   int
+		zebraNotifications  int
+	}{
+		{
+			desc:                "Output contains IPv4",
+			vtyshOutput:         peersCountersIPv4,
+			peer:                "172.18.0.4",
+			sessionUp:           1,
+			controlPacketInput:  5,
+			controlPacketOutput: 5,
+			echoPacketInput:     0,
+			echoPacketOutput:    0,
+			sessionUpEvents:     1,
+			sessionDownEvents:   0,
+			zebraNotifications:  4,
+		},
+		{
+			desc:                "Output contains IPv6",
+			vtyshOutput:         peersCountersIPv6,
+			peer:                "fc00:f853:ccd:e793::4",
+			sessionUp:           1,
+			controlPacketInput:  10,
+			controlPacketOutput: 10,
+			echoPacketInput:     0,
+			echoPacketOutput:    0,
+			sessionUpEvents:     1,
+			sessionDownEvents:   0,
+			zebraNotifications:  4,
+		},
+	}
+	peersCountersIPv4 = `
+	[
+		{
+			"multihop":false,
+			"peer":"172.18.0.4",
+			"vrf":"default",
+			"interface":"eth0",
+			"control-packet-input":5,
+			"control-packet-output":5,
+			"echo-packet-input":0,
+			"echo-packet-output":0,
+			"session-up":1,
+			"session-down":0,
+			"zebra-notifications":4
+		}
+	]
+	`
+	peersCountersIPv6 = `
+	[
+		{
+			"multihop":false,
+			"peer":"fc00:f853:ccd:e793::4",
+			"local":"fc00:f853:ccd:e793::6",
+			"vrf":"default",
+			"interface":"eth0",
+			"control-packet-input":10,
+			"control-packet-output":10,
+			"echo-packet-input":0,
+			"echo-packet-output":0,
+			"session-up":1,
+			"session-down":0,
+			"zebra-notifications":4
+		}
+	]
+	`
+)
+
+func TestBFDCollect(t *testing.T) {
+	for _, test := range bfdTests {
+		t.Run(test.desc, func(t *testing.T) {
+			tmpl, err := template.New(test.desc).Parse(bfdMetricsTmpl)
+			if err != nil {
+				t.Errorf("expected no error but got %s", err)
+			}
+
+			var w bytes.Buffer
+			err = tmpl.Execute(&w, map[string]interface{}{
+				"Peer":                test.peer,
+				"SessionUp":           test.sessionUp,
```

Comment:
I'm not sure but I think the unit tests aren't really parsing this stat? (1 is the default)
Should we add something that covers that the collection from BFDPeers actually work?